src/main/java/edu/ucsb/cs156/frontiers/controllers/RosterStudentsController.java

```diff
@@ -209,16 +214,18 @@ public Map<String, String> uploadRosterStudentsCSV(
       for (String[] row : myEntries) {
         RosterStudent rosterStudent = fromCSVRow(row, sourceType);
         UpsertResponse upsertResponse = upsertStudent(rosterStudent, course, RosterStatus.ROSTER);
-        InsertStatus s = upsertResponse.insertStatus;
-        counts[s.ordinal()]++;
+        if (upsertResponse.insertStatus == InsertStatus.REJECTED) {
+          rejectedStudents.add(rosterStudent);
+        } else {
+          InsertStatus s = upsertResponse.insertStatus;
+          counts[s.ordinal()]++;
+        }
       }
     }
-    return Map.of(
-        "filename", file.getOriginalFilename(),
-        "message",
-            String.format(
-                "Inserted %d new students, Updated %d students",
-                counts[InsertStatus.INSERTED.ordinal()], counts[InsertStatus.UPDATED.ordinal()]));
+    return new LoadResult(
+        counts[InsertStatus.INSERTED.ordinal()],
+        counts[InsertStatus.UPDATED.ordinal()],
+        rejectedStudents);
```

Comment:
What if we did something like this:
```suggestion
    LoadResult loadResult = new LoadResult(
        counts[InsertStatus.INSERTED.ordinal()],
        counts[InsertStatus.UPDATED.ordinal()],
        rejectedStudents);
    if (rejectStudents.length() > 0) {
       throw new RejectedStudentsException(loadResult);
    }
    return loadResult;
```

Then we can add an Exception Handler in the ApiController to catch this, and format a nice error message for the end user.   I agree that it's not a common error message, but failing silently, when it comes to getting students into the course, sounds like... well, "not ideal" is how I'll phrase it, but I'm being "diplomatic" :-)

Reply:
I think we agree that throwing an exception is overkill in the long run.. that it would be better to simply process the output of the request on the client side and throw it up in a modal... and it occurs to me that building a mechanism for this could also be leveraged for the CSV team upload, where there are similar issues, and the same data structure... 

It might be worth it to just go ahead and build that, especially if it can be reused when we build the frontend for PR #316.

But I'll leave it up you.